checkresponse.go

```diff
@@ -12,30 +12,31 @@ type HTTPError struct {
 	Response   *http.Response
 	StatusCode int
 	Err        error
-
-	raw string
+	Raw        []byte
 }
 
 // ErrHTTP define an http error
 var ErrHTTP = errors.New("http error")
 
 func (e *HTTPError) Error() string {
 	var delimiter string
-	if e.raw != "" {
+	if len(e.Raw) != 0 {
 		delimiter = " - "
 	}
 	return fmt.Sprintf("%v %v: %d%s%+v",
 		e.Response.Request.Method,
 		e.Response.Request.URL,
 		e.Response.StatusCode,
 		delimiter,
-		e.raw,
+		string(e.Raw),
 	)
 }
 func (e *HTTPError) Unwrap() error {
 	return e.Err
 }
 
+//func (e *HTTPError) Unmarshal
```

Comment:
We could set an `Unmarshal` method on the HTTPError struct as an utility to unmarshal the response error and a `GetRaw` method to geet the error as slice of bytes.
In this way, we could maintain the raw field private

Reply:
I would keep `Raw` available anyway because it may be useful to the user in the case in which the response is malformed and it's not `application/json`. 
I would add `GetJRawAsJSON` (or kind of) as a simple utility function.

Reply:
Ok, I agree. Could you add the method in this PR?